### PR TITLE
Feature - Schedule Management

### PR DESCRIPTION
**Pull request by v-stamenova**
This branch introduces the feature of managing the schedule. It is accessed currently only through route (/schedule/overview). Things that should be tested (I suggest fresh migrate and seed)
- [x] Only accessed when user is with role content moderator
- [x] Numbers for a) Presentations awaiting approval and b) Presentations awaiting to be scheduled are correct
- [x] When autofilling for the first time the moderator is required to generate the timeslots (will be asked for starting time of the conference, ending time of the conference, lunch break)
- [x] Validation for timeslots is working
- [x] The moderator can manually assign rooms and timeslots
- [x] The moderator can edit the already scheduled presentations (the rooms and timeslots)
- [x] There aren't any presentations that when scheduled are overlapping! - the autofill shouldn't be overlapping them and the manual assignment of room and timeslot should not be allowing an overlap
- [x] The timeslot reset is working correctly - it deletes all timeslots, redirects user to generate timeslots
- [x] The maximum participants numbers are correct - the lower of the two numbers: max participants of room and max participants of presentation

### CODE REVIEW

File: resources/views/moderator/schedule/timeslots-create.blade.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                    <div class="col-span-6 sm:col-span-4 py-4">
+                        <x-label for="ending" value="Ending time"/>
+                        <x-input id="ending" name="ending" type="time" value="{{old('ending')}}"
+                                 class="mt-1 block w-full"/>
+                        <x-input-error for="ending" class="mt-2"/>
+                    </div>
+                    <hr>
+                    <h3 class="text-lg text-gray-900 dark:text-white pt-5">Lunch break</h3>
+                    <div class="col-span-6 sm:col-span-4 pt-5">
+                        <x-label for="breakStart" value="Starting time of lunch break (default is 12:30)"/>
```
Comment (TimKardol) [inline resources/views/moderator/schedule/timeslots-create.blade.php:36]: When saving but not adding the time for lunchbreak, expected behaviour would be that it saves and then uses said default values. However it errors out with: The break start field must match the format H:i

File: resources/views/components/schedule-block.blade.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            eaque, exercitationem praesentium nihil.', 0, 55) . '...' }}
+        </p>
+    </div>
+    <div class="px-2 pb-2">
+        <span class="inline-block bg-gray-200 rounded-full px-3 py-1 text-xs font-semibold text-gray-700 mr-2 mb-2">difficulty</span>
+        <span class="inline-block bg-gray-200 rounded-full px-3 py-1 text-xs font-semibold text-gray-700 mr-2 mb-2">
+            {{$presentation->room->name}}
+        </span>
+        <span class="inline-block bg-gray-200 rounded-full px-3 py-1 text-xs font-semibold text-gray-700 mr-2 mb-2">
+            Max participants: {{$presentation->maxParticipants()}}
```
Comment (TimKardol) [inline resources/views/components/schedule-block.blade.php:21]: Instead of max participants, show how many spots are left open.

**Review by TimKardol — CHANGES_REQUESTED** [review]
Not fully done yet with the review. But these are some things I noticed at a first lookover.

File: resources/views/moderator/schedule/index.blade.php [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            <div class="grid grid-cols-8 gap-4 pb-12">
+                <div style="grid-column: span 3;">
+                    <h2 class="text-2xl text-gray-800 dark:text-gray-200 leading-tight">
+                        Current version of schedule
+                    </h2>
+                </div>
+                <div>
+                    <a href="{{route('moderator.requests', 'presentations')}}"
+                       class="bg-purple-800 text-xs text-white py-2 px-4 rounded block text-center transition-all duration-300 transform hover:scale-105">
+                        {{$numberOfPresentationRequest}}<br>Presentations<br>waiting to be approved
```
Comment (TimKardol) [inline resources/views/moderator/schedule/index.blade.php:23]: Using `<br>` is the old way of doing it. Preferred way is using `<div>` or `<p>` I think tailwind has some classes for lineheight as well. You can also leave this be for a frontend refactor round.

**Review by TimKardol — CHANGES_REQUESTED** [review]

File: resources/views/livewire/room-and-timeslot-selector.blade.php [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                        <option>No timeslots available for this room</option>
+                    </select>
+                @endif
+            @endif
+        </div>
+        <button class="bg-indigo-800 hover:bg-indigo-700 text-white py-2 px-4 rounded">
+            Save
+        </button>
+        @if($selectedRoom)
+            <p class="text-sm text-gray-900 dark:text-gray-200">This means that the presentation's maximum participants is {{$maxParticipants}}</p>
```
[the remark this answers is not in the document]
Comment (v-stamenova) [reply resources/views/livewire/room-and-timeslot-selector.blade.php:58]: The dropdown says the max participants for the room, but when we designed the db Daan and I decided to put max participants for room and max participants per presentation (given by the user who requested the presentation). We decided that whichever is the smallest of the two numbers is going to override the bigger number and the smallest number will stay as max participants. So when a room is selected, this shows whichever is the smaller number of max participants - the one that is for the presentation or the one for the room.

**Review by TimKardol — APPROVED** [review]
Thank you!

**Pull request merged**